tests/core/test_api.py

```diff
@@ -214,3 +215,47 @@ def mock_get_db_health(url: str) -> str:
     monkeypatch.setattr(main, "get_db_health", mock_get_db_health)
     response = test_client.get(test_config.cli.server_url + API_PREFIX + "/health")
     assert response.status_code == expected_status_code
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "sample_string", ["test1:'as?/<>.,", "!@#$%^&*()-_+=", "1a2b3c4d5e6F7G8H9I0J"]
+)
+def test_obscure_string(sample_string: str) -> None:
+    obscured_string = obscure_string(sample_string)
+    assert sample_string == unobscure_string(obscured_string)
+
+
+EXTERNAL_CONFIG_BODY = {
+    "aws": {
+        "region_name": getenv("AWS_DEFAULT_REGION", ""),
+        "aws_access_key_id": obscure_string(getenv("AWS_ACCESS_KEY_ID", "")),
+        "aws_secret_access_key": obscure_string(getenv("AWS_SECRET_ACCESS_KEY", "")),
+    }
+}
+
+
+@pytest.mark.external
+@pytest.mark.parametrize("generate_type, generate_target", [("systems", "aws")])
+def test_generate(
+    test_config: FidesctlConfig,
+    generate_type: str,
+    generate_target: str,
+    test_client: TestClient,
+) -> None:
+
+    data = {
+        "organization_key": "default_organization",
+        "generate": {
+            "config": EXTERNAL_CONFIG_BODY[generate_target],
+            "target": generate_target,
+            "type": generate_type,
+        },
+    }
+
+    response = test_client.post(
+        test_config.cli.server_url + API_PREFIX + "/generate/",
+        headers=test_config.user.request_headers,
+        data=dumps(data),
+    )
+    assert response.status_code == 200
```

Comment:
Since we know there will be at least some systems, should we also check that the results array is not empty?

Reply:
Originally I wanted to return a 204 here for items where we connected but no systems were found. My thinking was that one of the risk items is using a role with incorrect permissions to view things, but that could be sort of invalid ?

Reply:
An empty list is fine if no infrastructure is found in the aws account I'd say. I am curious about that use case though, i think some of the aws will fail while others will exclude resources. 

For this comment here though, I just meant to do an additional check for `len(response.generate_results) > 1` in this test

Reply:
Ya I got lost in the files, my bad :) makes sense here, good call 👍🏽